packages/wallets/src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import moment from 'moment';
+import Calendar, { CalendarProps } from 'react-calendar';
+import CalendarIcon from '../../public/images/ic-calendar.svg';
+import FlowTextField, { TFlowFieldProps } from '../FlowField/FlowTextField';
+import unixToDateString, { customFormatShortWeekday } from './utils';
+import 'react-calendar/dist/Calendar.css';
+import './DatePicker.scss';
+
+interface TDatePickerProps extends TFlowFieldProps {
+    mobileAlignment?: 'above' | 'below';
+    onDateChange: (formattedDate: string | null) => void;
+    variant?: 'dateOfBirth' | 'expiry';
+}
+
+const DatePicker = ({
+    defaultValue,
+    label,
+    message,
+    mobileAlignment = 'below',
+    name,
+    onDateChange,
+    validationSchema,
+    variant = 'dateOfBirth',
+}: TDatePickerProps) => {
+    const [selectedDate, setSelectedDate] = useState<Date | null>(defaultValue ? new Date(defaultValue) : null);
+    const [isCalendarOpen, setIsCalendarOpen] = useState<boolean>(false);
+    const datePickerRef = useRef<HTMLDivElement>(null);
+
+    const maxDate = variant === 'expiry' ? undefined : moment().subtract(18, 'years').toDate();
+    const minDate = variant === 'expiry' ? moment().add(2, 'days').toDate() : moment().subtract(100, 'years').toDate();
```

Comment:
Instead make maxDate and minDate as props. Maybe put these max/min date for dateofbirth and documentexpirydate in constants for accounts usage

Reply:
Updated 👍